sa/model.go

```diff
@@ -1294,3 +1295,239 @@ func setReplacementOrderFinalized(ctx context.Context, db db.Execer, orderID int
 	}
 	return nil
 }
+
+type pausedModel struct {
+	RegistrationID  int64      `db:"registrationID"`
+	IdentifierType  uint8      `db:"identifierType"`
+	IdentifierValue string     `db:"identifierValue"`
+	PausedAt        *time.Time `db:"pausedAt"`
+	UnpausedAt      *time.Time `db:"unpausedAt"`
+}
+
+// checkPairsPaused returns a list of identifierValues that are currently paused
+// for the given (account, hostname) pairs. The list will contain at most 15
+// values. If no pairs are currently paused, an empty list is returned.
+func checkPairsPaused(ctx context.Context, dbs db.Selector, regID int64, identifierType uint8, identifierValues []string) ([]string, error) {
+	if len(identifierValues) == 0 {
+		// No identifier values to check.
+		return []string{}, nil
+	}
+
+	query := fmt.Sprintf(`
+        SELECT identifierValue
+        FROM paused
+        WHERE 
+            registrationID = ? AND
+            identifierType = ? AND
+            identifierValue IN (%s) AND
+            unpausedAt IS NULL
+        LIMIT 15`,
```

Comment:
I don't think we need to limit this here. Presumably the case where this gets called is during new Order creation, in which case we already know that we're limited to at most 100 entries thanks to our names-per-cert limit.

I think this limit is more necessary for the GetPausedIdentifiersForAccount method I propose in the top-level comment.

Reply:
LIMIT 15 limits the number of potential matches before the transaction returns, not the number of order names to be checked. This is the list of paused names (for this account) to be included in the "User Action Required" message. This is done in an effort to keep the query time shorter but mostly in an effort to keep the log line reasonable for both ourselves (log storage) and for the Subscriber (short and easy to read).

Reply:
> Right, my thought was that an implicit LIMIT 100 (because there will be at most 100 input arguments)

Relying on a limit enforced somewhere far removed from this method is a strategy which  might easily invite future mishaps.

> I'm not worried about the performance of returning 100 rows, but maybe I should be?

Would it be fine? Probably. Though, this query _is_ right in the path of every New Order so I'd like to keep it as tight as possible.

Reply:
I decided to do most of this but return a message for metrics purposes at the end.